packages/react/src/index.ts

```diff
@@ -97,30 +153,31 @@ Object.defineProperties(Signal.prototype, {
 
 // Track the current dispatcher (roughly equiv to current component impl)
 let lock = false;
-const UPDATE = () => ({});
 let currentDispatcher: ReactDispatcher;
+
 Object.defineProperty(internals.ReactCurrentDispatcher, "current", {
 	get() {
 		return currentDispatcher;
 	},
-	set(api) {
+	set(api: ReactDispatcher) {
 		currentDispatcher = api;
 		if (lock) return;
 		if (api && !isInvalidHookAccessor(api)) {
-			// prevent re-injecting useReducer when the Dispatcher
-			// context changes to run the reducer callback:
+			// prevent re-injecting useMemo & useSyncExternalStore when the Dispatcher
+			// context changes.
 			lock = true;
-			const rerender = api.useReducer(UPDATE, {})[1];
+
+			const store = api.useMemo(createEffectStore, Empty);
+
+			useSyncExternalStore(
+				store.subscribe,
+				store.getSnapshot,
+				store.getSnapshot
```

Comment:
I believe the third argument is optional and defaults to the 2nd here?

```suggestion
				store.getSnapshot
```

Reply:
I don't remember if it warns when it fallbacks to the 1st one 🤔 
I'll check that and remove it if it's not needed.